core-api/src/test/java/com/optimizely/ab/config/parser/DefaultConfigParserTest.java

```diff
@@ -16,15 +16,76 @@
  */
 package com.optimizely.ab.config.parser;
 
+import com.optimizely.ab.config.ProjectConfig;
+import com.optimizely.ab.internal.PropertyUtils;
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static com.optimizely.ab.config.DatafileProjectConfigTestUtils.*;
+import static com.optimizely.ab.config.DatafileProjectConfigTestUtils.validConfigJsonV4;
 
 /**
  * Tests for {@link DefaultConfigParser}.
  */
+@RunWith(Parameterized.class)
 public class DefaultConfigParserTest {
 
+    @Parameterized.Parameters(name = "{index}")
+    public static Collection<Object[]> data() throws IOException {
+        return Arrays.asList(new Object[][]{
+            {
+                validConfigJsonV2(),
+                validProjectConfigV2()
+            },
+            {
+                validConfigJsonV3(),
+                validProjectConfigV3()
+            },
+            {
+                validConfigJsonV4(),
+                validProjectConfigV4()
+            }
+        });
+    }
+
+    @Parameterized.Parameter(0)
+    public String validDatafile;
+
+    @Parameterized.Parameter(1)
+    public ProjectConfig validProjectConfig;
+
+    /**
+     * This method is to test DefaultConfigParser when different default_parser gets set.
+     * For example: when optimizely_default_parser environment variable will be set to "com.google.gson.Gson" than
+     * "DefaultConfigParser.getInstance()" returns "GsonConfigParser" and parse ProjectConfig using it. Also
+     * this test will assertThat "configParser" is instance of "GsonConfigParser.class"
+     *
+     * @throws Exception
+     */
     @Test
-    public void createThrowException() throws Exception {
-        // FIXME - mdodsworth: hmmm, this isn't going to be the easiest thing to test
+    public void testPropertyDefaultParser() throws Exception {
+        String defaultParser = PropertyUtils.get("default_parser");
+        ConfigParser configParser = DefaultConfigParser.getInstance();
+        ProjectConfig actual = configParser.parseProjectConfig(validDatafile);
+        ProjectConfig expected = validProjectConfig;
+        verifyProjectConfig(actual, expected);
+        if(defaultParser != null) {
+            if (DefaultConfigParser.ConfigParsers.GSON_CONFIG_PARSER.toString().equals(defaultParser)) {
```

Comment:
with the enum as defined earlier we can use `ConfigParserSupplier.valueOf(defaultParser)`.